### ISSUE TITLE
onNotification callback handler not getting called in iOS for remote notifications

### ISSUE THREAD

**Issue by nairabhijit** [zo0r/react-native-push-notification#1103]
This happens when the application is in a killed state and I receive a notification and if i click on it, the onNotification callback handler is never called. This works when the application is in active state. I tried everything and gone through all issues similar to this posted by people on github but without any progress, also I rechecked notification configuration in Xcode. Any help would be appreciated.

**Comment by matt-wright86** [comment]
@nairabhijit is onNotification called after you tap on the notification?

**Comment by nairabhijit** [comment]
No it doesn't and this is what the issue is.

**Comment by maamounapprise** [comment]
Have you enabled push notification capability in xcode?

**Comment by markuscosinus** [comment]
Do you get this message, when you click on Notifications a few times?
`Multiple notification open events received before the JS notifications module has been initialized`.
Then this might be related to #1026

**Comment by nairabhijit** [comment]
I resolved the issue by following this [link](https://github.com/facebook/react-native/issues/8580). I have to do a lot of code rewrites, I had to replace libraray wix react-native-navigation with react-navigation, as I wasn't knowing how to set the appProperties  to rootView in AppDelegate.m file with react-native-navigation. Also I figured out later that on release environment I am able to get the notification in `PushNotiifcation.getInitialNotification()` but not with debug environment. The behaviour is very inconsistent. Nevertheless now I am able to get the notification in bebug as well as release envrioment when the application is in killed state.
